src/modules/claim/claim-verification.service.spec.ts

```diff
@@ -0,0 +1,148 @@
+  resolveServiceEndpoints: jest.fn(),
+};
+
+const MockIssuerVerificationService = {
+  verifyIssuer: jest.fn(),
+};
+
+const MockRoleCredentialResolver = {
+  getCredential: jest.fn(),
+};
+
+describe('ClaimVerificationService', () => {
+  let service: ClaimVerificationService;
+  let module: TestingModule;
+  beforeEach(async () => {
+    module = await Test.createTestingModule({
+      providers: [
+        ClaimVerificationService,
+        {
+          provide: DIDService,
+          useValue: MockDidService,
+        },
+        {
+          provide: Logger,
+          useValue: MockLogger,
+        },
+        {
+          provide: IssuerVerificationService,
+          useValue: MockIssuerVerificationService,
+        },
+        {
+          provide: RoleCredentialResolver,
+          useValue: MockRoleCredentialResolver,
+        },
+      ],
+    }).compile();
+    await module.init();
+    service = module.get<ClaimVerificationService>(ClaimVerificationService);
+  });
+
+  afterEach(async () => {
+    jest.resetAllMocks();
+    module.close();
+  });
+
+  it('service should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  describe('verifyEnrolmentPreconditions', () => {
+    it('should thrown an error if a precondition type is a type other than "role"', async () => {
+      const enrolmentPreconditions = [
+        { type: PreconditionType.Role, conditions: ['conditionone.org.ewc'] },
+        {
+          type: 'someType' as PreconditionType,
+          conditions: ['conditionone.org.ewc'],
+        },
+      ];
+      try {
+        const result = await service.verifyEnrolmentPreconditions(
+          enrolmentPreconditions,
+          resolvedCredential.payload.did,
+          'claimType'
+        );
+        expect(result).toThrowError(
+          'Error: An enrolment precondition has an unsupported precondition type. Supported precondition types include: "Role"'
+        );
+      } catch (_) {}
+    });
+    it('should throw an error if the resolveCredentialAndVerify returns an error', async () => {
+      const mockError = {
+        isVerified: false,
+        errors: [`Issuer not authorised to issue the credential`],
+      };
+      jest
+        .spyOn(service, 'resolveCredentialAndVerify')
+        .mockResolvedValueOnce(mockError);
+      const enrolmentPreconditions = [
+        { type: PreconditionType.Role, conditions: ['conditionone.org.ewc'] },
+      ];
+      try {
+        const result = await service.verifyEnrolmentPreconditions(
+          enrolmentPreconditions,
+          resolvedCredential.payload.did,
+          'claimType'
+        );
+        expect(result).toThrowError(
+          `Role enrolment precondition not met for user: ${
+            resolvedCredential.payload.did
+          } for role: ${'conditionone.org.ewc'}. Verification errors for enrolment preconditions: ${JSON.stringify(
+            mockError.errors
+          )}`
+        );
+      } catch (_) {}
```

Comment:
why is the test in this a `try/catch` block? I thought that `toThrowError` would handle the thrown error. I'm worried that a try/catch block could swallow unexpected errors and cause the test to pass despite this error.